### PR TITLE
decouple_wallet_from_subtensor

### PR DESCRIPTION
**Pull request by unconst**
--- removes the wallet from the subtensor client, now wallet is passed to relevant functions.

### CODE REVIEW

File: bittensor/executor.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def __init__(
         self.wallet = wallet
 
         if subtensor == None:
-            subtensor = bittensor.subtensor.Subtensor( config = self.config, wallet = self.wallet )
+            subtensor = bittensor.subtensor.Subtensor( config = self.config )
```
Comment (dehekkendekrekker) [inline bittensor/executor.py:64]: Is there a reason to assume subtensor == None?

File: bittensor/metagraph.py [hunk 2]
```diff
@@ -634,7 +634,7 @@ def set_weights(self, weights: torch.FloatTensor, wait_for_inclusion = False, ti
         code, message = self._try_emit(weights, wait_for_inclusion, timeout)
         if code == Metagraph.EmitSuccess:
             # ---- Emit was a success. ----
-            logger.info("Emission was successful and entered the block.")
+            logger.success("Emission was successful and entered the block.")
```
Comment (dehekkendekrekker) [inline bittensor/metagraph.py:637]: Dope

File: bittensor/subtensor.py [hunk 3]
```diff
@@ -230,7 +230,7 @@ def connection_error_message():
             # ---- Get next endpoint ----
             ws_chain_endpoint = self.endpoint_for_network( blacklist = attempted_endpoints )
```
Comment (dehekkendekrekker) [inline bittensor/subtensor.py:231]: This logic can be abstracted out

**Review by dehekkendekrekker — APPROVED** [review]
Good PR. It's small enough to not get lost in complexity, and clearly shows the intention.
However, the proposed changes, even though this could work, are a bit hacky.

This client class was intended to be a low-level class, basically only interfacing with the subtensor server. Some higher level objects have been introduced (the config object and the wallet object) being container objects for data that is used throughout the codebase. 

This PR removes the injection of the wallet object from the constructor, and injects them into the methods. This is not necessary. Injection into the constructor is a valid approach, as the contents of the objects do not change. It's just that another layer is needed that is able to funnel the data  specific to the client from the config and wallet objects into client's constructor.

Especially in combination with the logic that looks for an appropriate endpoint to use for the client class, a factory class could be a good solution to this problem.

Given a service class that determines the endpoint to use, the factory could simply derive a client object using the wallet and the config as a parameter.

This effectively decouples the code, making it easier to test, without having to resort to cumbersome mocking strategies. (For the client and factory class that is, the service class is a different case).

I'll approve this PR, as to not stall the process, but I do recommend implementing the proposed structure.

File: bittensor/executor.py (hunk 1 above)
Comment (unconst) [reply bittensor/executor.py:64]: You can instantiate the executor with a subtensor client, otherwise it just uses the config file.

**Pull request merged**